src/program.ts

```diff
@@ -4533,6 +4533,25 @@ export class Class extends TypedElement {
     }
     return out;
   }
+
+  getVisitPrototype(): DeclaredElement | null {
+    const visitPrototype = this.getMember("__visit");
+    if (visitPrototype == null) {
+      return null;
+    }
+    if (this.isDeclaredInLibrary) {
+      return visitPrototype;
+    }
+    const decorator = visitPrototype.declaration.decorators;
+    if (!decorator || (decorator && decorator.every((v) => v.decoratorKind != DecoratorKind.UNSAFE))) {
+      return null;
+    }
```

Comment:
As I understand currently if `@unsafe` doesn't exists than rewriting of `__visit` silently doesn't happening. Will be great add new diagnostic error message that would describe that this operation is unsafe and requires the explicit addition of the "@unsafe" decorator. And the user has to decide whether to add it or go some other way

Reply:
In old logic, we can only rewrite `__visit` in `~/lib`. 
What I want to do is using `@unsafe` to enable rewriting `__visit` if this class is in the outside. For example, some class in assembly/index.ts

Reply:
Yes, I understand this. I'm suggesting also fix current behaviour when you try to do the same but without `@unsafe` decorator (in main branch it just silently ignored)